src/variables.jl

```diff
@@ -1391,7 +1458,7 @@ function add_variable(
     var_indices = _moi_add_constrained_variables(
         backend(model),
         variable.scalar_variables,
-        variable.set,
+        model_convert(model, variable.set),
```

Comment:
This is inconsistent with the constraint case where the `model_convert` is called by the macro and not inside `add_constraint`.
So it should be applied to
https://github.com/jump-dev/JuMP.jl/blob/8c205287191a345eb85db096deb845114a9ee38f/src/macros.jl#L2740
like in
https://github.com/jump-dev/JuMP.jl/blob/8c205287191a345eb85db096deb845114a9ee38f/src/macros.jl#L1054-L1058

Reply:
Done